src/Microsoft.DotNet.Wpf/src/WindowsBase/MS/Internal/IO/Packaging/XmlDigitalSignatureProcessor.cs

```diff
@@ -874,9 +874,7 @@ private static AsymmetricAlgorithm GetPrivateKey(X509Certificate2 cert)
             // Get[Algorithm]PrivateKey methods would always have returned the private key if the PrivateKey property would
             // But Get[Algorithm]PrivateKey methods never throw but returns null in case of error during cryptographic operations
             // But we want exception to be thrown when an error occurs during a cryptographic operation so that we can revert the changes
-            #pragma warning disable SYSLIB0028
-            return cert.PrivateKey;
-            #pragma warning restore SYSLIB0028
+            throw new CryptographicException("PrivateKey is not retrievable.");
```

Comment:
You are now throwing an exception if all three of the Get*Key methods return null.
1. Does `cert.PrivateKey` also throw in that case?   I couldn't find a doc that guaranteed that.
2. Does it throw the same exception, with the same message?   If not, this is a breaking change.
3. Does the message need to be localized?   If it's visible to users (i.e. not caught by WPF code), then yes.
4. The comment at the head of the method says the keys returned by PrivateKey are different from Get*Key.  If answer to (1) is 'no', any code that acts on that difference is now obsolete.  Is there such code?  (Not visible in this review.)

Reply:
https://docs.microsoft.com/en-us/dotnet/api/system.security.cryptography.x509certificates.x509certificate2.privatekey?view=net-5.0. 

For 1, there is no specific language about a null key throwing, but it does say an invalid RSA, DSA, or 'unreadable' key will throw CryptographicException.  One indication that it throws if everything is null is that the callers like PackageDigitalSignatureManager.Sign catch but do not handle the null case.  

For 2/3, we can leave the 509certificate2.privatekey replacement for RC2 and check in everything else.  Or if 509certificate2.privatekey can only throw a single error code in CryptographicException when all three Get*Key methods return null, then it can be replaced now.  

For 4, I read the comment as the PrivateKey property was used as a replacement for the Get*PrivateKey methods to intentionally cause an exception, rather than returning null.

Reply:
That doc doesn't mention ECDsa keys at all, so I'm not sure whether PrivateKey can be an ECDsa key at all, and if so whether c.PrivateKey throws.  The doc could be literally true, or simply out-of-date since ECDsa was added.  Have you or Ashish asked the crypto team?

In any case, if we called c.PrivateKey to cause a user-visible exception, we are now throwing that exception ourselves and for compat we must throw the same exception (we do, if doc is correct) with the same message (unlikely), and that message needs localization.  Ask the crypto team what message they use, and use SR to localize it.  Plenty of examples in our code.